### PR TITLE
Add doc on troubleshooting campaigns

### PR DESCRIPTION
**Pull request by mrnugget**
This goes hand in hand with https://github.com/sourcegraph/sourcegraph/pull/16748 and can hopefully help @sourcegraph/customer-engineering to help customers debug campaigns :)

### CODE REVIEW

File: doc/campaigns/references/troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+```bash
+src login https://sourcegraph.example.com
+```
+
+If `src` is correctly configured, then the output should look similar to the following:
+
+```
+
+✔️  Authenticated as mmy-username on https://ssourcegraph.example.com
```
Comment (eseliger) [inline doc/campaigns/references/troubleshooting.md:23]:
```suggestion
✔️  Authenticated as my-username on https://sourcegraph.example.com
```

**Review by eseliger — APPROVED** [review]

File: doc/campaigns/references/troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
+# Troubleshooting
+
+This page is meant to help narrow down and eliminate problems when trying to execute a campaign spec with `src campaign [apply|preview]` or managing an already created campaign and its changesets.
+
+## Executing campaign steps
+
+Since `src campaign [apply|preview]` execute a campaign spec on the host machine on which it is executed (and not on the Sourcegraph instance), there are a lot of different possibilities that can cause it to fail: from missing dependencies to missing credentials when trying to connect ot the Sourcegraph instance.
+
+The following questions can be used to find out what's causing the problem and should ideally be answered with "yes".
```
Comment (LawnGnome) [inline doc/campaigns/references/troubleshooting.md:9]: I like this approach. :+1:

File: doc/campaigns/references/troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+Go through the "[Quickstart](../quickstart.md)" to run a campaign that adds "Hello World" to `README.md` files with the following campaign spec:
+
+```yaml
+name: hello-world
+description: Add Hello World to READMEs
+
+# Find all repositories that contain a README.md file.
+on:
+  - repositoriesMatchingQuery: file:README.md
```
Comment (LawnGnome) [inline doc/campaigns/references/troubleshooting.md:57]: Given how many repositories most Sourcegraph instances will have, we might want to add a `count:` directive to keep this speedy.

File: doc/campaigns/references/troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+Once you know that executing the campaign spec works with one or more repositories, you can expand the scope back to its original form and hopefully nail down in which repository execution breaks.
+
+### Can you download a repository archive?
+
+If `src` is stuck in the "Initializing workspace" phase for a repository or fails to initialize the workspace, try to see whether you can download an archive of the repository manually on your command line with `curl`:
+
+```
+curl -L -v -X GET -H 'Accept: application/zip' \
+  -H "Authorization: token $SRC_ACCESS_TOKEN" \
+  "SRC_ENDPOINT/github.com/my-org/my-repo@refs/heads/master/-/raw" \
```
Comment (LawnGnome) [inline doc/campaigns/references/troubleshooting.md:120]:
```suggestion
  "$SRC_ENDPOINT/github.com/my-org/my-repo@refs/heads/master/-/raw" \
```

**Review by LawnGnome — APPROVED** [review]
Awesome. Straight up awesome.

File: doc/campaigns/references/troubleshooting.md (hunk 3 above)
Comment (mrnugget) [inline doc/campaigns/references/troubleshooting.md:57]:
```suggestion
# Find one repository with a README.md
on:
  - repositoriesMatchingQuery: repohasfile:README.md count:1
```

Comment (mrnugget) [reply doc/campaigns/references/troubleshooting.md]: done! Did something different, but has the same effect.

**Review by LawnGnome — APPROVED** [review]

**Pull request merged**